### PR TITLE
Chloe/1.3 rooms

### LINKED ISSUE #21: As an OHS admin I can edit a room's type
so that I can log upgrades like installation of aircon.

### LINKED ISSUE #116: Bug: Help Command doesn't work for aliases
Steps to reproduce
- Type `alias c/rooms a/r`
- Type `help r`

Expected behaviour: the help message for rooms should be shown
Actual behaviour: `Cannot show help for r...`

### LINKED ISSUE #128: Add commands for adding, deleting and editing rooms

### PR DESCRIPTION
**Pull request by chloelee767**
### What this does
1. Removes the CommandMapper and replaces it with the CommandWordEnum as the single source of truth for binding commands together with their parsers and help messages 
1. Allows users to add, delete and edit rooms using the `add-room`, `delete-room` and `edit-room` commands. (closes #128)
    - Adding and deleting of rooms is only allowed in the following scenarios: when there is no `resireg.json` file, after clearing all the rooms using `clear`, or starting a new semester after running `archive`. In these scenarios, the app is said to be in "new" mode. While in the new mode, users can only view and modify rooms (and not students or anything else). 
    - While using new mode, users can finalize the rooms in the building and go to normal mode using the `finalize` command.
    - Rooms can be edited at any time, but only tags and the room type can be edited (otherwise users can effectively create or remove rooms outside of the new mode)
1. Allocations have also been removed from the sample ResiReg so that users can delete any of the rooms while in new mode
1. Command words and prefixes have been made more consistent.
1. Also fixes #116 

#### Some limitations / things to take note about the new features
- If you exit ResiReg while in the new mode, the next time you start up ResiReg, it will be in normal mode.
- `undo` and `redo` operations can be used to undo/redo the `finalize` command
- `help` will only show the help for the commands available in the current mode
- Rooms are not binnable

#### What still needs to be done
- Tests for all new commands and the implementation of the different modes for the app.
- Update user guide and developer guide

### How to test

Things you should test:
- Adding, deleting and editing rooms.
- Only the correct commands are allowed in new mode and normal mode.
- Undo-ing and redo-ing `finalize`

### Notes for other devs
- The way to bind commands has changed (yet again) from CommandMapper + CommandWordEnum to only CommandWordEnum
- The files to change to add new tabs has changed (yet again, sorry (":): instead of `MainPanel.java` and `MainPanel.fxml`, please modify `NormalMainPanel.java` and `NormalMainPanel.fxml`

### CODE REVIEW

File: src/main/java/seedu/resireg/model/alias/AliasUtils.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * Utility functions for classes in the model.alias package.
+ */
+public class AliasUtils {
+    /**
+     * Converts a list of {@code CommandWordAlias} to a map of alias to command word.
+     */
+    public static Map<String, String> makeAliasToCommandWordMap(List<CommandWordAlias> aliasList) {
+        return aliasList.stream().collect(Collectors.toMap(
+            commandWordAlias -> commandWordAlias.alias.alias,
```
Comment (mkeoliya) [inline src/main/java/seedu/resireg/model/alias/AliasUtils.java]: Can we change this to `commandWordAlias.alias.toString()` to follow the Law of Demeter?

**Review by mkeoliya — COMMENTED** [review]
Just one small edit from my end

File: src/main/java/seedu/resireg/MainApp.java [hunk 2]
```diff
@@ -76,10 +76,13 @@ public void init() throws Exception {
     private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
         Optional<ReadOnlyResiReg> resiRegOptional;
         ReadOnlyResiReg initialData;
+        boolean allowRoomCreation = true;
```
Comment (JingYenLoh) [inline src/main/java/seedu/resireg/MainApp.java]: Also a nit on my part, should this be `isRoomCreationAllowed` following CS2103's conventions?

File: src/main/java/seedu/resireg/logic/LogicManager.java [hunk 3]
```diff
@@ -72,6 +78,16 @@ public CommandResult execute(String commandText) throws CommandException, ParseE
         return commandResult;
     }
 
+    private void addAliases(Map<String, Parser<Command>> map, List<CommandWordAlias> aliasList) {
+        for (CommandWordAlias alias : aliasList) {
+            String commandWord = alias.getCommandWord().commandWord;
+            String aliasString = alias.getAlias().alias;
```
Comment (JingYenLoh) [inline src/main/java/seedu/resireg/logic/LogicManager.java]: Similar to what Mayank said, perhaps a minor change can be made to obey CS2103's Law of Demeter?

File: src/main/resources/view/NormalMainPanel.fxml [hunk 4]
```diff
@@ -11,7 +11,7 @@
 
     <TabPane fx:id="tabPane"
              tabClosingPolicy="UNAVAILABLE">
-        <!-- Students and rooms tabs will be created programatically -->
+        <!-- Students and rooms tabs will be created programmatically -->
```
Comment (JingYenLoh) [inline src/main/resources/view/NormalMainPanel.fxml:14]: 🤣

**Review by JingYenLoh — CHANGES_REQUESTED** [review]
Wow, great stuff. Can you adjust the nitpicks though (I should do the same for my Semester model @.@)?

File: src/main/java/seedu/resireg/model/alias/AliasUtils.java (hunk 1 above)
Comment (chloelee767) [reply src/main/java/seedu/resireg/model/alias/AliasUtils.java]: Alright changed it, thanks for the catch

File: src/main/java/seedu/resireg/MainApp.java (hunk 2 above)
Comment (chloelee767) [reply src/main/java/seedu/resireg/MainApp.java]: Alright changed it, thanks for the catch

File: src/main/java/seedu/resireg/logic/LogicManager.java (hunk 3 above)
Comment (chloelee767) [reply src/main/java/seedu/resireg/logic/LogicManager.java]: Yup, changed it

File: src/main/resources/view/NormalMainPanel.fxml (hunk 4 above)
Comment (chloelee767) [reply src/main/resources/view/NormalMainPanel.fxml:14]: What can I say, IntelliJ suggested it :woman_shrugging:

**Review by JingYenLoh — APPROVED** [review]

**Pull request merged**